packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -538,14 +547,24 @@ class BallisticScrollActivity extends ScrollActivity {
 
   late AnimationController _controller;
 
+  final double _initVelocity;
+
+  final double _initPosition;
+
   @override
   void resetActivity() {
     delegate.goBallistic(velocity);
   }
 
   @override
   void applyNewDimensions() {
-    delegate.goBallistic(velocity);
+    final Simulation? newSimulation = delegate.updateBallisticAnimation(
+      _initVelocity,
+      _initPosition,
+    );
+    if (newSimulation != null) {
```

Comment:
From original PR: Similar concern as above, what does this look like when the animation speeds are vastly different?

Source: https://github.com/flutter/flutter/pull/100133/files#r844476769